feat(r/slos): get all slos per environment (#662)
honeycombio/terraform-provider-honeycombio: docs/data-sources/slos.md

# Data Source: honeycombio_slos

> The SLOs data source retrieves the SLOs of a dataset, with the option of narrowing the retrieval by providing a `detail_filter`.  To retrieve all of the SLOs in an environment pass the special parameter `__all__` in place of a dataset.

Author: I wonder if it's worth adding a note that you'll need an env api key for this?

The following arguments are supported:

> * `dataset` - (Optional) The dataset to retrieve the SLOs list from. If omitted, all SLOs in the environment will be returned.

Reviewer: slight adjustment as the filter can be used so it won't necessarily be "all"
```suggestion
* `dataset` - (Optional) The dataset to retrieve the SLOs list from. If omitted, the lookup will be Environment-wide.
```
